bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLSequence.java

```diff
@@ -551,6 +551,16 @@ public void freezeDirect() {
         }
     }
 
+    @Override
+    public synchronized boolean isFrozen() {
```

Comment:
IIUC, a sequence cannot be updated. So if all the children are read-only that indirectly makes the sequence also read-only?

If that is the case, if the following check evaluates to true, can't we set the frozen flag to true here? That way in the read-only scenario, we can avoid going through all the children over and over again?

Reply:
Membership of the sequence is unmodifiable but each individual xml item is not.

Reply:
Yeah, but if this evaluates to true once, it will always be true after that right?

Reply:
Ah yes, I miss that on your first comment.